CIP-0062/README.md

```diff
@@ -0,0 +1,417 @@
+Status: Draft
+Type: Standards
+Created: 2021-06-11
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This document describe the interface between webpage / web-based stack and cardano wallets. This specifies that API of the javascript object that need to be injected into the web applications in order to support all the Governance features.
+
+These definitions extend [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/) to provide specific support for vote delegation.
+
+# Motivation
+
+The goal for this CIP is to extend the dApp-Wallet web bridge to enable the construction of transactions containing metadata that conforms to
+[CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format - Updated)](https://cips.cardano.org/cips/cip36/),
+enabling new functionality including vote delegation to either private or public representatives (dReps),
+splitting or combining of private votes,
+the use of different voting keys or delegations
+for different purposes (Catalyst etc).
+
+# Specification
+
+## Version
+
+The API Extension specified in this document will count as version 0.2.0 for version-checking purposes below.
+
+## Data Types
+
+### PublicKey
+
+TODO: Define this.
+
+### GovernanceKey
+
+```js
+type GovernanceKey = {
+  votingKey: string,
+  weight: number
+}
+```
+
+* `votingKey` - Ed25519 pubkey 32 bytes HEX string
+* `weight` - Used to calculate the actual voting power using the rules described
+  in [CIP-36](https://cips.cardano.org/cips/cip36/).
+
+### Voting Purpose
+
+```js
+type enum VotingPurpose = {
+  CATALYST = 0
+}
+```
+
+`VotingPurpose`: Defines the voting purpose of the governance functions. This is used
+to limit the scope of the governance functions. For example, a voting purpose
+might be a subset of Catalyst proposals, a council election, or even some
+private purpose (agreed by convention). Currently, only voting purpose 0 (Zero)
+is defined, and it is for Catalyst events.
+
+**IMPORTANT**: The List of purposes in this CIP should be considered the
+authoritative list of known purposes, subject to future amendment. Other voting
+purposes will be defined as required, by either an update to this CIP or a
+future CIP listing currently allocated Voting Purposes.
+
+### BlockDate
+
+```js
+interface BlockDate {
+  epoch: number
+  slot: number
+}
+```
+
+* epoch - an epoch value on the voting block chain.
+* slot - a block slot number on the voting block chain.
+
+#### Proposal
+
+```js
+interface Proposal {
+  votePlanId: number
+  voteOptions: number[]
+}
+```
+
+Proposal information.
+
+* votePlanId - The unique ID of this proposal in the vote plan.  This will be the same as anchored on voting chain, and is managed by the dApp.
+* voteOptions - List of possible "choices" which can be made for this proposal.
+
+### Vote
+
+```js
+interface Vote {
+  proposal: Proposal,
+  choice: number,
+  expiration: BlockDate
+  purpose: VotingPurpose,
+  spendingCounter: number
+}
```

Comment:
Need to ensure the vote can be declared as either private/encrypted or public because signing the vote differs between them.